algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -54,11 +56,26 @@ namespace Kokkos {
 namespace Experimental {
 namespace Impl {
 
+template <typename ViewType, typename Enabled = void>
+struct has_random_access_iterator : std::false_type {};
```

Comment:
with me suggested change above, we don't need this and it can stay inside the randomAccessIterator class